units/src/lib.rs

```diff
@@ -1,22 +1,55 @@
 // SPDX-License-Identifier: CC0-1.0
 
-//! # Rust Bitcoin - unit types
+//! Basic (numeric) types used by the Rust Bitcoin ecosystem.
 //!
-//! This library provides basic types used by the Rust Bitcoin ecosystem.
+//! This crate contains basic types that have minimal requirements on the platform they run on.
+//! Specifically, they do not require an allocator and they do not require `usize` to be at least
+//! 32-bit. If you need more than this crate provides check the [`bitcoin`] crate or the
+//! [`bitcoin-primitives`] crate.
 //!
-//! If you are using `rust-bitcoin` then you do not need to access this crate directly. Everything
-//! here is re-exported in `rust-bitcoin` at the same path. Also the same re-exports exist in
-//! `primitives` if you are using that crate instead of `bitcoin`.
+//! # Guidance on crate use
+//!
+//! Libraries that only need the types present in this crate should depend only on this crate, not
+//! `bitcoin` or `bitcoin-primitives` so that they don't add bloat to compilation and review. It is
+//! recommended that binaries or other root crates depend on `bitcoin` during the prototyping stage
+//! and then optionally try to trim down the dependencies by using the leaf crates. However this is
+//! unlikely to be feasible for non-trivial applications.
+//!
+//! If you are using the `bitcoin` crate then you do not need to access this crate directly.
+//! Everything here is re-exported in `bitcoin` at the same path. Also the same re-exports exist in
+//! `bitcoin-primitives` if you are using that crate instead of `bitcoin`.
```

Comment:
I rekon this paragraph would be better at the top of the section. Perhaps split like this?

 ```suggestion
//! If you are using the `bitcoin` crate then you do not need to access this crate directly.
//!
//! Everything here is re-exported in `bitcoin` at the same path. Also the same re-exports exist in
//! `bitcoin-primitives` if you are using that crate instead of `bitcoin`.
```

Reply:
Good idea, I think I'll use italic as well.